src/SFML/Window/DRM/DRMContext.cpp

```diff
@@ -130,6 +134,306 @@ namespace
         initialized = false;
     }
 
+    void drmFbDestroyCallback(gbm_bo* bo, void* data)
+    {
+        int drmFd = gbm_device_get_fd(gbm_bo_get_device(bo));
+        DrmFb* fb = static_cast<DrmFb*>(data);
+
+        if (fb->fbId)
+            drmModeRmFB(drmFd, fb->fbId);
+
+        delete fb;
+    }
+
+    DrmFb* drmFbGetFromBo(gbm_bo& bo)
+    {
+        int drmFd = gbm_device_get_fd(gbm_bo_get_device(&bo));
+        DrmFb* fb = static_cast<DrmFb*>(gbm_bo_get_user_data(&bo));
+        if (fb)
+            return fb;
+
+        fb = new DrmFb();
+        fb->bo = &bo;
+
+        const sf::Uint32 width = gbm_bo_get_width(&bo);
+        const sf::Uint32 height = gbm_bo_get_height(&bo);
+        const sf::Uint32 format = gbm_bo_get_format(&bo);
+
+        sf::Uint32 strides[4] = {0};
+        sf::Uint32 handles[4] = {0};
+        sf::Uint32 offsets[4] = {0};
+        uint64_t modifiers[4] = {0};
+        modifiers[0] = gbm_bo_get_modifier(&bo);
+        const int num_planes = gbm_bo_get_plane_count(&bo);
+        for (int i = 0; i < num_planes; ++i)
+        {
+            strides[i] = gbm_bo_get_stride_for_plane(&bo, i);
+            handles[i] = gbm_bo_get_handle(&bo).u32;
+            offsets[i] = gbm_bo_get_offset(&bo, i);
+            modifiers[i] = modifiers[0];
+        }
+
+        sf::Uint32 flags = 0;
+        if (modifiers[0])
+        {
+            flags = DRM_MODE_FB_MODIFIERS;
+        }
+
+        int ret = drmModeAddFB2WithModifiers(drmFd, width, height, format, handles, strides, offsets, modifiers, &fb->fbId, flags);
+
+        if (ret)
+        {
+            std::memset(handles, 0, 16);
+            handles[0] = gbm_bo_get_handle(&bo).u32;
+            std::memset(strides, 0, 16);
+            strides[0] = gbm_bo_get_stride(&bo);
+            std::memset(offsets, 0, 16);
+            ret = drmModeAddFB2(drmFd, width, height, format, handles, strides, offsets, &fb->fbId, 0);
+        }
+
+        if (ret)
+        {
+            sf::err() << "Failed to create fb: " << std::strerror(errno) << std::endl;
+            delete fb;
+            return NULL;
+        }
+
+        gbm_bo_set_user_data(&bo, fb, drmFbDestroyCallback);
+
+        return fb;
+    }
+
+    sf::Uint32 findCrtcForEncoder(const drmModeRes& resources, const drmModeEncoder& encoder)
```

Comment:
What's a Crtc?

Reply:
https://events.static.linuxfound.org/sites/events/files/slides/brezillon-drm-kms.pdf

<img width="1706" alt="Screenshot 2022-11-27 at 3 15 30 PM" src="https://user-images.githubusercontent.com/39244355/204162503-4fa8f0b3-7291-449b-b9ad-041d2956e8c5.png">

Looks like `CrtController` is a more apt name.

Reply:
CRTC is DRM slang, so to say. I suggest you read the kernel documentation, https://docs.kernel.org/gpu/drm-kms.html#crtc-abstraction is a good source explaining what a CRTC is. Scroll up to the top for a short presentation of DRM. There is a number of other acronyms that are related to DRM, and should always be kept to report the DRM naming